### DISCUSSION TITLE
Convert graph to CSC

### DISCUSSION THREAD

**Discussion by atale480** [pyg-team/pytorch_geometric#7317] [Q&A]
I'm trying to use CuGraphSAGEConv Layer. [Here](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.conv.CuGraphSAGEConv.html) it says that I need CSC representation of graph, and for that use `CuGraphModule.to_csc()` method to convert an edge_index representation to the desired format. I'm working on 3D mesh data. Which contain positions and faces information. How to convert this information or the edge index information to CSC?

I came across one method called `to_torch_csc_tensor`. Does this outputs same that is needed as input to CuGraphSAGEConv layer? If not then what is correct way to get data in CSC format as per need of CuGraphSAGEConv.

Also when I try to initialize CuGraphSAGEConv as `L = CuGraphSAGEConv( in_channels = 3, out_channels = 16 )` it gives error as `ModuleNotFoundError: 'CuGraphSAGEConv' requires 'pylibcugraphops'`. 
I'm not sure what does this error mean. Any help would be great regarding this issues. Thanks

**Comment by rusty1s** [comment]
Since you are working on mesh data, you can directly use the `FaceToEdge` transform. Afterwards, you can call `CuGraphModule.to_csc()` from the given `edge_index` representation. There is nothing to do with `pos`. You can just use it as input features into `CuGraphSAGEConv`.

You also need to install `pylibcugraphops` from `conda`, see https://anaconda.org/nvidia/pylibcugraphops.
